pkg/cmd/experimental/diagnostics/config.go

```diff
@@ -23,6 +25,9 @@ func (o DiagnosticsOptions) detectClientConfig() (bool, []types.DiagnosticError,
 func (o DiagnosticsOptions) buildRawConfig() (*clientcmdapi.Config, error) {
 	kubeConfig, configErr := o.Factory.OpenShiftClientConfig.RawConfig()
 	if len(kubeConfig.Contexts) == 0 {
+		if configErr == nil {
+			configErr = errors.New("No contexts found in config file.")
+		}
 		return nil, configErr
 	}
 	return &kubeConfig, configErr
```

Comment:
That way down here, you can simply do `return &kubeConfig, nil` so its very clear to readers that this is the success path.